datasets/us/nj/med_exclusions/crawler.py

```diff
@@ -0,0 +1,76 @@
+from datetime import datetime
+from typing import Dict
+from rigour.mime.types import PDF
+
+from zavod import Context, helpers as h
+from zavod.shed.gpt import run_image_prompt
+
+prompt = """
+Extract structured data from the following page of a PDF document. Return 
+a JSON list (`providers`) in which each object represents an medical provider.
+Distinct record rows alternate between grey and white backgrounds. The name
+of a provider might be split onto the next row within a record.
+Each object should have the following fields: `provider_name`, `title`,
+`npi_number`, `street`, `city`, `state`, `zip`, `action`, `effective_date`,
+`expiration_date`.
+Return an empty string for unset fields.
+"""
+
+
+def crawl_item(row: Dict[str, str], context: Context):
+
+    address = h.make_address(
+        context,
+        street=row.pop("street"),
+        city=row.pop("city"),
+        state=row.pop("state"),
+        country_code="US",
+        postal_code=row.pop("zip"),
+    )
+
+    if not row.get("title"):
+        entity = context.make("Company")
+        entity.id = context.make_id(row.get("npi_number"), row.get("provider_name"))
+        entity.add("name", row.pop("provider_name"))
+    else:
+        entity = context.make("Person")
+        entity.id = context.make_id(row.get("npi_number"), row.get("provider_name"))
+        h.apply_name(entity, full=row.pop("provider_name"))
+        entity.add("title", row.pop("title"))
+
+    entity.add("npiCode", row.pop("npi_number"))
+    entity.add("country", "us")
+
+    sanction = h.make_sanction(context, entity)
+    h.apply_date(sanction, "startDate", row.pop("effective_date"))
+    sanction.add("provisions", row.pop("action"))
+
+    if row.get("expiration_date") and row.get("expiration_date").upper() not in [
+        "PERMANENT",
+        "DECEASED",
+    ]:
+        is_debarred = (
+            datetime.strptime(row.get("expiration_date"), "%m/%d/%Y") > datetime.today()
+        )
```

Comment:
let's rather parse with h.extract_date - a new function that's actually used by apply_date so it would produce the same result.

Reply:
or better, use the approach I used on nevada.